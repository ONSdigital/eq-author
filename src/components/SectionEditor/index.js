import React from "react";
import PropTypes from "prop-types";
import CustomPropTypes from "custom-prop-types";

import Field from "components/Forms/Field";
import SeamlessInput from "components/SeamlessInput/SeamlessInput";
import SeamlessTextArea from "components/SeamlessTextArea/SeamlessTextArea";
import withEntityEditor from "components/withEntityEditor";

class SectionEditor extends React.Component {
  render() {
    const { section, onUpdate, onChange } = this.props;

    return (
      <div id="section-editor">
        <Field id="title">
          <SeamlessInput
            placeholder="Section title"
            size="medium"
            value={section.title}
            onChange={onChange}
            onBlur={onUpdate}
          />
        </Field>
        <Field id="description" optional>
          <SeamlessTextArea
            cols="30"
            rows="5"
            placeholder="Enter a description (optional)â€¦"
            value={section.description}
            onChange={onChange}
            onBlur={onUpdate}
          />
        </Field>
      </div>
    );
  }
}

SectionEditor.propTypes = {
  section: CustomPropTypes.section.isRequired,
  onChange: PropTypes.func.isRequired,
  onUpdate: PropTypes.func.isRequired
};

export default withEntityEditor("section")(SectionEditor);
