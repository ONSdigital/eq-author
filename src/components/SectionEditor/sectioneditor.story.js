import React from "react";
import { set, merge } from "lodash";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import styled from "styled-components";
import SectionEditor from "./index";
import { MemoryRouter, Route } from "react-router";
import { ApolloProvider } from "react-apollo";

const CenterXY = styled.div`
  position: absolute;
  height: 20em;
  width: 20em;
  top: 50%;
  left: 50%;
  transform: translateY(-50%) translateX(-50%);
`;

const sectionQuery = {
  sections: [
    {
      id: "1",
      title: "",
      pages: [
        {
          id: "1",
          title: "",
          answers: [
            {
              id: "1",
              label: "",
              type: "Currency"
            }
          ]
        }
      ]
    }
  ]
};

const query = () => ({
  questionnaire: {
    sections: [
      {
        id: "1",
        title: "",
        pages: [
          {
            id: "1",
            title: "",
            answers: [
              {
                id: "1",
                label: "",
                type: "Currency"
              }
            ]
          }
        ]
      }
    ]
  }
});

const client = {
  query: query,
  readQuery: query
};

const props = {
  onUpdate: action("onUpdate"),
  client
};

class SectionEditorWrapper extends React.Component {
  constructor() {
    super();
    this.state = {
      "section.title": "Section title",
      "section.description": "Section description"
    };
  }
  handleChange = ({ name, value }) => {
    const newState = set(merge({}, this.state), name, value);
    this.setState(newState);
  };

  render() {
    const renderStory = () => (
      <SectionEditor
        onChange={this.handleChange}
        section={sectionQuery}
        sectionTitle={this.state["section.title"]}
        sectionDescription={this.state["section.description"]}
        {...props}
      />
    );
    return (
      <ApolloProvider client={props.client}>
        <MemoryRouter
          initialEntries={[
            {
              pathname: "/questionnaire/1/design/1/1"
            }
          ]}
          initialIndex={0}
        >
          <Route
            path="/questionnaire/:questionnaireId/design/:sectionId/:pageId"
            exact={false}
            render={renderStory}
          />
        </MemoryRouter>
      </ApolloProvider>
    );
  }
}

const CenterDecorator = storyFn => <CenterXY>{storyFn()}</CenterXY>;

storiesOf("SectionEditor", module)
  .addDecorator(CenterDecorator)
  .add("Default", () => <SectionEditorWrapper />);
