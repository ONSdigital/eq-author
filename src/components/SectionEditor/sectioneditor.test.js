import React from "react";
import { mount } from "enzyme";
import SectionEditor from "./index";

describe("SectionEditor", () => {
  const render = component =>
    <component
      onChange={jest.fn()}
      sectionTitle="Title"
      sectionDescription="Description"
    />;

  it("should render", () => {
    expect(render(SectionEditor)).toMatchSnapshot();
  });
});
