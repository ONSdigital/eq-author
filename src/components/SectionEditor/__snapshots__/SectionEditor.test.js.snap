// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionEditor should render 1`] = `
<SectionEditor__SectionCanvas
  data-test="section-editor"
  id={null}
>
  <DeleteConfirmDialog
    alertText="All questions in this section will also be removed. This may affect piping and routing rules elsewhere."
    data-test="dialog-delete-confirm"
    icon="icon-dialog-section.svg"
    isOpen={false}
    onClose={[Function]}
    onDelete={[Function]}
  />
  <MoveSectionQuery
    questionnaireId="1"
  />
  <SectionEditor__Padding>
    <SectionEditor__Alias>
      <SectionEditor__AliasField>
        <Label
          bold={true}
          htmlFor="section-alias"
        >
          Section short code (optional)
        </Label>
        <withChangeHandler(WrappingInput)
          autoFocus={true}
          data-test="section-alias"
          id="section-alias"
          maxLength={255}
          name="alias"
          onBlur={[Function]}
          onChange={[Function]}
        />
      </SectionEditor__AliasField>
      <CharacterCounter
        limit={24}
      />
    </SectionEditor__Alias>
  </SectionEditor__Padding>
  <SectionEditor__Padding>
    <RichTextEditor
      autoFocus={false}
      controls={
        Object {
          "emphasis": true,
        }
      }
      id="title"
      label="Title"
      multiline={false}
      onUpdate={[Function]}
      placeholder=""
      size="large"
      testSelector="txt-section-title"
      value="Section 1"
    />
  </SectionEditor__Padding>
</SectionEditor__SectionCanvas>
`;

exports[`SectionEditor should render 2`] = `
<SectionEditor__SectionCanvas
  data-test="section-editor"
  id={null}
>
  <DeleteConfirmDialog
    alertText="All questions in this section will also be removed. This may affect piping and routing rules elsewhere."
    data-test="dialog-delete-confirm"
    icon="icon-dialog-section.svg"
    isOpen={false}
    onClose={[Function]}
    onDelete={[Function]}
  />
  <MoveSectionQuery
    questionnaireId="1"
  />
  <SectionEditor__Padding>
    <SectionEditor__Alias>
      <SectionEditor__AliasField>
        <Label
          bold={true}
          htmlFor="section-alias"
        >
          Section short code (optional)
        </Label>
        <withChangeHandler(WrappingInput)
          autoFocus={true}
          data-test="section-alias"
          id="section-alias"
          maxLength={255}
          name="alias"
          onBlur={[Function]}
          onChange={[Function]}
        />
      </SectionEditor__AliasField>
      <CharacterCounter
        limit={24}
      />
    </SectionEditor__Alias>
  </SectionEditor__Padding>
  <SectionEditor__Padding>
    <RichTextEditor
      autoFocus={false}
      controls={
        Object {
          "emphasis": true,
        }
      }
      id="title"
      label="Title"
      multiline={false}
      onUpdate={[Function]}
      placeholder=""
      size="large"
      testSelector="txt-section-title"
      value="Section 2"
    />
  </SectionEditor__Padding>
</SectionEditor__SectionCanvas>
`;
